backend/dataall/core/environment/services/managed_iam_policies.py

```diff
@@ -19,17 +22,54 @@ def __init__(
         self.environmentUri = environmentUri
         self.resource_prefix = resource_prefix
 
+    @abstractmethod
+    def generate_policy_name(self) -> str:
+        """
+        Returns string and needs to be implemented in the ManagedPolicies inherited classes
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def generate_empty_policy(self) -> dict:
+        """
+        Returns dict and needs to be implemented in the ManagedPolicies inherited classes
+        """
+        raise NotImplementedError
+
+
+class PolicyManager(object):
+    def __init__(
+            self,
+            role_name,
+            account,
+            environmentUri,
+            resource_prefix,
+    ):
+        self.role_name = role_name
+        self.account = account
+        self.environmentUri = environmentUri
+        self.resource_prefix = resource_prefix
+        self.ManagedPolicies = ManagedPolicy.__subclasses__()
+        logger.info(f'Found {len(self.ManagedPolicies)} subclasses of ManagedPolicy')
+        self.initializedPolicies = [self._initialize_policy(policy) for policy in self.ManagedPolicies]
+
+    def _initialize_policy(self, managedPolicy):
+        return managedPolicy(
+            role_name=self.role_name,
+            account=self.account,
+            environmentUri=self.environmentUri,
+            resource_prefix=self.resource_prefix
+        )
+
     def create_all_policies(self, managed) -> bool:
         """
         Manager that registers and calls all policies created by data.all modules and that
         need to be created for consumption roles and team roles
         """
-
-        policies = ManagedPolicy.__subclasses__()
-        logger.info(f'Found {len(policies)} subclasses of ManagedPolicy')
-        for policy in policies:
-            empty_policy = policy.generate_empty_policy(self)
-            policy_name = policy.generate_policy_name(self)
+        for Policy in self.initializedPolicies:
```

Comment:
why `Policy` and not `policy`? this might confuse someone thinking that this is a PolicyClass object (similar to the rest of the places you used capital P)

Reply:
It is a Policy class, that's why I defined it with uppercase